plenum/server/replica.py

```diff
@@ -2634,19 +2651,22 @@ def process_requested_prepare(self, prepare: Prepare, sender: str):
     def process_requested_commit(self, commit: Commit, sender: str):
         return self._process_requested_three_phase_msg(commit, sender, self.requested_commits)
 
-    def is_pre_prepare_time_correct(self, pp: PrePrepare) -> bool:
+    def is_pre_prepare_time_correct(self, pp: PrePrepare, sender: str) -> bool:
         """
         Check if this PRE-PREPARE is not older than (not checking for greater
         than since batches maybe sent in less than 1 second) last PRE-PREPARE
         and in a sufficient range of local clock's UTC time.
         :param pp:
         :return:
         """
+        tpcKey = (pp.viewNo, pp.ppSeqNo)
         return ((self.last_accepted_pre_prepare_time is None or
```

Comment:
It would be great to simplify this methods as follows:
```
if self.last_accepted_pre_prepare_time is not None and pp.ppTime < self.last_accepted_pre_prepare_time:
    return False
if tpcKey not in self.pre_prepare_tss:
    return False
....
```

Reply:
Ok